File: src/ast/treesitter/parsers/tests/cases/ts/person.ts
/**
 * This is a class representing a Person.
 */
class Person {
    private name: string;
    private age: number;

    /**
     * This is the constructor method for the Person class.
     * @param name The name of the person.
     * @param age The age of the person.
     */
    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }

    /**
     * This method returns the name of the person.
     * @returns The name of the person.
     */
    getName(): string {
        return this.name;
    }

    /**
     * This method returns the age of the person.
     * @returns The age of the person.
     */
    getAge(): number {
        return this.age;
    }

    /**
     * This method sets the name of the person.
     * @param name The new name of the person.
     */
    setName(name: string): void {
        this.name = name;
    }

    /**
     * This method sets the age of the person.
     * @param age The new age of the person.
     */
    setAge(age: number): void {
        this.age = age;
    }
}

File: refact-agent/engine/src/ast/treesitter/parsers/tests/cases/java/person.java
/**
 * Represents a simple model of a Person.
 */
public class Person {

    // Fields
    private String name;
    private int age;

    /**
     * Constructs a new Person with the specified name and age.
     *
     * @param name the name of the person
     * @param age  the age of the person
     */
    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }
    
    public abstract void a();
    
    /**
     * Gets the name of the person.
     *
     * @return the name of the person
     */
    public String getName() {
        return name;
    }

    /**
     * Sets the name of the person.
     *
     * @param name the new name of the person
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Gets the age of the person.
     *
     * @return the age of the person
     */
    public int getAge() {
        return age;
    }

    /**
     * Sets the age of the person.
     *
     * @param age the new age of the person
     */
    public void setAge(int age) {
        this.age = age;
    }

    /**
     * Returns a string representation of the person.
     *
     * @return a string representation of the person
     */
    @Override
    public String toString() {
        return "Person{name='" + name + "', age=" + age + "}";
    }

    public static void main(String[] args) {
        Person person = new Person("John Doe", 30);
        System.out.println(person);
    }
}